File: src/main/java/mediathek/tool/SwingErrorDialog.java
package mediathek.tool;

import mediathek.config.Konstanten;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;
import java.io.PrintWriter;
import java.io.StringWriter;

public class SwingErrorDialog {
    public static void showExceptionMessage(@NotNull Component parentComponent,
                                            @NotNull String messageText,
                                            @NotNull Exception exception) throws HeadlessException {

        StringWriter stringWriter = new StringWriter();
        exception.printStackTrace(new PrintWriter(stringWriter));

        JLabel message = new JLabel(messageText);
        message.setBorder(BorderFactory.createEmptyBorder(3, 0, 10, 0));

        JTextArea text = new JTextArea();
        text.setEditable(false);
        text.setFont(UIManager.getFont("Label.font"));
        text.setText(stringWriter.toString());
        text.setCaretPosition(0);

        JScrollPane scroller = new JScrollPane(text);
        scroller.setPreferredSize(new Dimension(640, 350));

        JPanel panel = new JPanel();
        panel.setLayout(new BorderLayout());

        panel.add(message, BorderLayout.NORTH);
        panel.add(scroller, BorderLayout.SOUTH);

        JOptionPane.showMessageDialog(parentComponent, panel, Konstanten.PROGRAMMNAME,
                JOptionPane.ERROR_MESSAGE);

    }
}

File: src/main/kotlin/mediathek/tool/SwingErrorDialog.kt
/*
 * Copyright (c) 2026 derreisende77.
 * This code was developed as part of the MediathekView project https://github.com/mediathekview/MediathekView
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package mediathek.tool

import mediathek.config.Konstanten
import java.awt.BorderLayout
import java.awt.Component
import java.awt.Dimension
import java.awt.HeadlessException
import java.io.PrintWriter
import java.io.StringWriter
import javax.swing.*

object SwingErrorDialog {
    @Throws(HeadlessException::class)
    fun showExceptionMessage(parentComponent: Component?, messageText: String, exception: Throwable) {
        val stringWriter = StringWriter()
        exception.printStackTrace(PrintWriter(stringWriter))

        val message = JLabel(messageText).apply {
            border = BorderFactory.createEmptyBorder(3, 0, 10, 0)
        }

        val text = JTextArea().apply {
            isEditable = false
            font = UIManager.getFont("Label.font")
            text = stringWriter.toString()
            caretPosition = 0
        }

        val scroller = JScrollPane(text).apply {
            preferredSize = Dimension(640, 350)
        }

        val panel = JPanel().apply {
            layout = BorderLayout()
            add(message, BorderLayout.NORTH)
            add(scroller, BorderLayout.SOUTH)
        }

        JOptionPane.showMessageDialog(parentComponent, panel, Konstanten.PROGRAMMNAME, JOptionPane.ERROR_MESSAGE)
    }
}
